services/streammanager/kafka/kafkamanager.go

```diff
@@ -226,8 +231,13 @@ func Produce(jsonData json.RawMessage, producer interface{}, destConfig interfac
 
 	var config = Config{}
 	jsonConfig, err := json.Marshal(destConfig)
+	if err != nil {
+		return makeErrorResponse(err)
```

Comment:
In this and the next two occurances, failing jobs because of poor destination config structure or being unable to marshal/unmarshal would make us lose data and not  be able to fix issue. Lets panic instead

Reply:
done